app/src/main/java/protect/card_locker/preferences/SettingsActivity.java

```diff
@@ -166,6 +169,10 @@ private void refreshActivity(boolean reloadMain) {
         }
 
         @Override
+        @SuppressWarnings("deprecation") // setTargetFragment
+        // androidx.preference.PreferenceDialogFragmentCompat uses the deprecated method
+        // `getTargetFragment()`, which throws if `setTargetFragment()` isn't used before.
+        // While this isn't fixed on upstream, suppress the deprecation warning
```

Comment:
Do you have a link to some issue or bugtracker showing this is an upstream bug, so in the future checking when this deprecation warning suppression can be removed would be easier?

Reply:
Added an upstream link to the bug